wsproto/__init__.py

```diff
@@ -5,17 +5,27 @@
 
 A WebSocket implementation.
 """
+from enum import Enum
+
 from .connection import ConnectionType
-from .handshake import H11Handshake
+from .handshake import  H2Handshake, H11Handshake
 
 __version__ = "0.12.0+dev"
 
 
+class HTTPVersion(Enum):
+    H11 = 1
+    H2 = 2
+
+
 class WSConnection(object):
-    def __init__(self, connection_type):
+    def __init__(self, connection_type, http_version=HTTPVersion.H11):
         # type: (ConnectionType) -> None:
         self.client = connection_type is ConnectionType.CLIENT
-        self.handshake = H11Handshake(connection_type)
+        if http_version is HTTPVersion.H11:
+            self.handshake = H11Handshake(connection_type)
+        else:
+            self.handshake = H2Handshake(connection_type)
```

Comment:
How about directly passing in a Handshake object instead of adding another enum?

Assuming we could pass in `conn = WSConnection(... handshake=None)`, this would give us a nice way to disable handshakes and directly get a "ready-to-exchange-WS-frames" WSConnection object. Thoughts?

Reply:
I'm happy to pass a `handshake_class` rather than the enum, but I think the `handshake=None` won't work as you hope, however the `Connection` class itself is exactly a "ready-to-exchange-WS-frames" object. The reason is that the Connection object must know what extensions to use on initialization.

Reply:
Right, I remember we already talked about it.
Please add a good doc-string that explains the two different use cases - then I'm happy with it.